### PR TITLE
Travis npm publishing

### PR DESCRIPTION
**Pull request by AmeeMorris**
So this should give us some lovely auto deployments, see docs here: https://docs.travis-ci.com/user/deployment/npm/

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
- - yarn test:ci
- - yarn lint
+- yarn test:ci
+- yarn lint
 cache: yarn
+deploy:
+  provider: npm
+  email: dev@gathercontent.com
+  api_key:
+    secure: nLmB2qOw2/n6Uvb2/ubaeKDvx+vBdFNEnG+KmJdaNz51c0G1NQ5+MZv1ukmTTdr6Q1GmzGz2L7ZHJoRUtuUj3eqqJPaVc/NrZmmEpq771lm0+C8UtN98XvIq4A/Fb4jjbg3QLUM+RNLT4NGFF8kGHVd5Zo6VUNu6JUMFGSVvqDDth34l8zB4uy7GWJVxZj8iNHEbrlOxJdZtJ5F01QQjXlr3xXkiYMG8Uz4sIVFfqX0G7jE8+oZG6LmKO06B9sreBb2xlrlQGAF9hiLkyTLFN9sFczr5MP843tnPGLkWcN2/H783mZQsGoZgHesygRaRCfTjxFEchWEsS7AQiALxKwTxDGXv52HPhOrx+cNgAYxDX+B+JPs+sMi6kBHNva6y8q0NzqJ4kxOUEusvT8KVL/fQmtLdQzvYxt5EgfDpVReGcAOHfQ6wIaMcXXKRqzD+kaknxWQzwcmPLhI7NSxYZLgFNUIoYZdtJt4PEAWHEiyq07i4v8LgsErqtesC5gokjNUqCAZq9PYyxg/MW13sE0zjm3JH7FUfZzra8OKDN7Os6v5gbtWxSGmYJCVEbv6Ay7uV5lQlBdXo+gg0ex/1dw9SeDJAkWq4OMGX7OEtaPjgg5NJlf+Z0q9VJZ/6UALdPtbWmaWCkd09KQwLO56UbA5uIBaUMGm6ypJzOutoqV8=
```
Comment (kyleharper) [inline .travis.yml]: How secure is it to add this into version control? 🤔

File: README.md [hunk 2]
```diff
@@ -48,10 +48,4 @@ require('gather-ui/dist/styles.css');
...
 ## Deploying the Library
- You will need to login with an NPM account that has access to the package
-```
-npm version  <major | minor | patch> // We usually go with patch, but minor or major may be required
-git push --tags
-git push origin <branch-name>
-npm publish
-```
\ No newline at end of file
+The library will automatically deploy when a new tag is pushed to the master branch. You can create a tag using the web interface, using the command line with [git tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging), or running `npm version  <major | minor | patch>` (We usually go with patch, but minor or major may be required) will also create a tag for you. Just ensure when pushing you use `git push --tags`.
```
Comment (kyleharper) [inline README.md]: I don't think we need to say that we usually patch.

File: .travis.yml (hunk 1 above)
Comment (DavidHollins6) [inline .travis.yml]: Love this! Just wondering if this key is a secret or can be public?

File: README.md (hunk 2 above)
Comment (AmeeMorris) [reply README.md]: Copied from old docs tbh!

**Comment by AmeeMorris** [comment]
@DavidHollins6 @kyleharper the key is encrypted (see https://docs.travis-ci.com/user/encryption-keys/ and https://docs.travis-ci.com/user/deployment/npm/#npm-auth-token)

Perhaps we can do this too: https://docs.npmjs.com/creating-and-viewing-authentication-tokens#cidr-restricted-token-errors 

Open to other suggestions too :)

**Comment by AmeeMorris** [comment]
@kyleharper also happy to mob about this, my knowledge is just the docs I've linked in the PR description, so it'l be good to have more minds on deck. 

As for tags & master confusion, if you look here: https://github.com/gathercontent/gather-ui/pull/482/files#diff-354f30a63fb0907d4ad57269548329e3R14 in our `on` arguments we have `tags` set to true, but also the `branch` set to `master`. I'm not 100% sure (requires testing) but I believe those two args will only trigger a publish when its a new tag on the master branch.

**Pull request merged**